ironfish/src/network/peerNetwork.ts

```diff
@@ -443,7 +445,10 @@ export class PeerNetwork {
 
     switch (style) {
       case RoutingStyle.gossip: {
-        this.gossipRouter.register(type, hdlr)
+        this.gossipRouter.register(
+          type,
+          hdlr as (message: IncomingGossipGeneric<T>) => Promise<boolean | void>,
```

Comment:
maybe `| boolean | void` on the end to match the other return types? or is that not needed now since onNewBlock is async

Reply:
Updated, added it so handlers in prod code or tests don't have to make it async if they don't need it to.